packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -400,6 +389,33 @@ export class Permissions {
     );
   };
 
+  public canCreateArchetype = (
+    archetype: Pick<ArchetypeInterface, "projects">
+  ): boolean => {
+    return this.checkProjectFilterPermission(
+      { projects: archetype?.projects ? archetype.projects : [] },
+      "manageArchetype"
+    );
+  };
+
+  public canUpdateArchetype = (
+    archetype: Pick<ArchetypeInterface, "projects">
+  ): boolean => {
```

Comment:
For update permission checks, you need to have two arguments: the old object and the new one (in case the list of projects is being changed).

Instead of `checkProjectFilterPermission`, you can use `checkProjectFilterUpdatePermission`.